### DISCUSSION TITLE
[@types/auth0] Auth0 library SignInToken type is not correct

### DISCUSSION THREAD

**Discussion by kchow1985** [DefinitelyTyped/DefinitelyTyped#59442] [Issues with a @types package] [answered]
Auth0 npm library's `SignInToken` is either not correct or outdated (best to confirm with a representative from Auth0)

the current type provided is
```
export interface SignInToken {
    access_token: string;
    id_token?: string;
    token_type?: string;
    expiry: number;
}
```

when I call the library's function which returns this type, the object is actually like this:
```
{
    access_token: string;
    id_token: string;
    token_type: string;
    expires_in: number;
}
```

the time for the token to expire is place in the property named `expires_in` instead of `expiry`

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
The docs about payload says:
https://github.com/auth0/node-auth0/blob/8fe9adebaf845243c1d3f9e548726836dc308758/src/auth/PasswordlessAuthenticator.js#L116
https://auth0.com/docs/authenticate/passwordless/implement-login/embedded-login/relevant-api-endpoints#post-oauth-token
```txt
HTTP/1.1 200 OK
Content-Type: application/json
{
"access_token":"eyJz93a...k4laUWw",
"refresh_token":"GEbRxBN...edjnXbL",
"id_token":"eyJ0XAi...4faeEoQ",
"token_type":"Bearer",
"expires_in":86400
}
```

**Comment by kchow1985** [comment]
Yes, the payload (the one you posted) is what I got from the `auth0` javascript library, `SignInToken` type from @types/auth0 has `expiry` instead of `expires_on`

Please check
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b7103222b46f60f46abc80763b49ae6d34107283/types/auth0/index.d.ts#L1010

**Comment by peterblazejewicz** [comment]
I can make a change, but that would require other owners review, thx!

**Comment by peterblazejewicz** [comment, marked as answer]
fixing #59479
